isso/js/app/lib/identicons.js

```diff
@@ -4,92 +4,91 @@
 
   Inspired by http://codepen.io/gschier/pen/GLvAy
 */
-define(["app/lib/promise", "app/config"], function(Q, config) {
-
-    "use strict";
-
-    // Number of squares width and height
-    var GRID = 5;
-
-    var pad = function(n, width) {
-        return n.length >= width ? n : new Array(width - n.length + 1).join("0") + n;
-    };
-
-    /**
-     * Fill in a square on the canvas.
-     */
-    var fill = function(svg, x, y, padding, size, color) {
-        var rect = document.createElementNS("http://www.w3.org/2000/svg", "rect");
-
-        rect.setAttribute("x", padding + x * size);
-        rect.setAttribute("y", padding + y * size);
-        rect.setAttribute("width", size);
-        rect.setAttribute("height", size);
-        rect.setAttribute("style", "fill: " + color);
-
-        svg.appendChild(rect);
-    };
-
-    /**
-     * Pick random squares to fill in.
-     */
-    var generateIdenticon = function(key, padding, size) {
-
-        var svg =  document.createElementNS("http://www.w3.org/2000/svg", "svg");
-        svg.setAttribute("version", "1.1");
-        svg.setAttribute("viewBox", "0 0 " + size + " " + size);
-        svg.setAttribute("preserveAspectRatio", "xMinYMin meet");
-        svg.setAttribute("shape-rendering", "crispEdges");
-        fill(svg, 0, 0, 0, size + 2*padding, config["avatar-bg"]);
-
-        if (typeof key === null) {
-            return svg;
-        }
+const Q = require("app/lib/promise");
+
+"use strict";
+
+// Number of squares width and height
+var GRID = 5;
+
+var pad = function(n, width) {
+    return n.length >= width ? n : new Array(width - n.length + 1).join("0") + n;
+};
 
-        Q.when(key, function(key) {
-            var hash = pad((parseInt(key.substr(-16), 16) % Math.pow(2, 18)).toString(2), 18),
-                index = 0;
+/**
+ * Fill in a square on the canvas.
+ */
+var fill = function(svg, x, y, padding, size, color) {
+    var rect = document.createElementNS("http://www.w3.org/2000/svg", "rect");
 
-            svg.setAttribute("data-hash", key);
+    rect.setAttribute("x", padding + x * size);
+    rect.setAttribute("y", padding + y * size);
+    rect.setAttribute("width", size);
+    rect.setAttribute("height", size);
+    rect.setAttribute("style", "fill: " + color);
 
-            var i = parseInt(hash.substring(hash.length - 3, hash.length), 2),
-                color = config["avatar-fg"][i % config["avatar-fg"].length];
+    svg.appendChild(rect);
+};
 
-            for (var x=0; x<Math.ceil(GRID/2); x++) {
-                for (var y=0; y<GRID; y++) {
+/**
+ * Pick random squares to fill in.
+ */
+var generateIdenticon = function(key, padding, size, config) {
```

Comment:
`config` should probably be  passed to this function on line 85 of this file

Reply:
In this particular instance you are right, but looking over this code again, it occurred to me that Isso currently doesn't use the `blank`/`generateBlank` functions at all.

I'll adjust this commit to let `generateBlank` use a passed `config` for the sake of keeping current functionality.

Reply:
Done